pkg/cluster/sync.go

```diff
@@ -1188,12 +1181,16 @@ func (c *Cluster) updateSecret(
 	} else {
 		// username might not match if password rotation has been disabled again
 		if secretUsername != string(secret.Data["username"]) {
-			*retentionUsers = append(*retentionUsers, secretUsername)
-			secret.Data["username"] = []byte(secretUsername)
-			secret.Data["password"] = []byte(util.RandomPassword(constants.PasswordLength))
-			secret.Data["nextRotation"] = []byte{}
-			updateSecret = true
-			updateSecretMsg = fmt.Sprintf("secret %s does not contain the role %s - updating username and resetting password", secretName, secretUsername)
+			if len(string(secret.Data["username"])) != len(secretUsername) {
```

Comment:
You already compare the user name strings in the if above, so no need to also compare the length? I assume same length but different content is equally wrong as different length.

Maybe you just need to add the else case which is not there in the current code.